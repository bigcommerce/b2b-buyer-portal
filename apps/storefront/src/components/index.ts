export { default as B3AddToQuoteTip } from './B3AddToQuoteTip'
export { default as B3Card } from './B3Card'
export { default as B3CollapseContainer } from './B3CollapseContainer'
export { default as B3CustomForm } from './B3CustomForm'
export { default as B3Dialog } from './B3Dialog'
export { default as B3GlobalTip } from './B3GlobalTip'
export * from './B3LinkTipContent'
export { default as B3ProductList } from './B3ProductList'
export { default as B3QuantityTextField } from './B3QuantityTextField'
export { default as B3Tag } from './B3Tag'
export { default as B3Tip } from './B3Tip'
export { default as CustomButton } from './button/CustomButton'
export { default as CheckoutTip } from './extraTip/CheckoutTip'
export { default as GlobalDialog } from './extraTip/GlobalDialog'
export * from './HeadlessController'
export * from './layout'
export { B3PageMask, Loading, showPageMask } from './loadding'
export { default as HomePageLoadding } from './loadding/B3HomePageLoadding'
export { default as B3HoverButton } from './outSideComponents/B3HoverButton'
export { default as B3MasquradeGobalTip } from './outSideComponents/B3MasquradeGobalTip'
export { default as RegisteredCloseButton } from './RegisteredCloseButton'
export { default as B3Sping } from './spin/B3Sping'
export { default as B3NoData } from './table/B3NoData'
export { B3PaginationTable } from './table/B3PaginationTable'
export { B3Table } from './table/B3Table'
export { default as ThemeFrame } from './ThemeFrame'
export { default as B3Upload } from './upload/B3Upload'
