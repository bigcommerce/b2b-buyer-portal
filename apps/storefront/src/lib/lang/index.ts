export { default as LangProvider } from './LangProvider';
export * from './useB3Lang';
