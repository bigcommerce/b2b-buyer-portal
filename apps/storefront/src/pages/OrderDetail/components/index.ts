export { default as DetailPagination } from './DetailPagination';
export { default as OrderAction } from './OrderAction';
export { default as OrderBilling } from './OrderBilling';
export { default as OrderHistory } from './OrderHistory';
export { default as OrderShipping } from './OrderShipping';
