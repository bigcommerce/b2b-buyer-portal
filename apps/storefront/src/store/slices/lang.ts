import { createSlice, PayloadAction } from '@reduxjs/toolkit'

export interface LangState {
  translations: Record<string, string>
}

const initialState: LangState = {
  translations: JSON.parse(localStorage.getItem('sf-translations') || '{}'),
}

const langSlice = createSlice({
  name: 'lang',
  initialState,
  reducers: {
    setTranslations: (
      state,
      { payload }: PayloadAction<LangState['translations']>
    ) => {
      state.translations = payload
    },
    updateTranslations: (
      state,
      { payload }: PayloadAction<LangState['translations']>
    ) => {
      state.translations = { ...state.translations, ...payload }
    },
  },
})

export const { setTranslations, updateTranslations } = langSlice.actions

export default langSlice.reducer
