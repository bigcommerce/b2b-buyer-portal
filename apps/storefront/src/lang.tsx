import { useContext, useState } from 'react';

import { LangProvider } from '@/lib/lang';

import { CustomStyleContext } from './shared/customStyleButton';

export function LangWrapper({ children }: { children: React.ReactNode }) {
  const {
    state: {
      loginPageButton: { createAccountButtonText, signInButtonText },
      loginPageDisplay: { pageTitle },
      masqueradeButton,
      addQuoteBtn,
      floatingAction,
      addToAllQuoteBtn,
      shoppingListBtn,
    },
  } = useContext(CustomStyleContext);
  const [customText] = useState<Record<string, string>>(
    Object.entries({
      'login.button.createAccount': createAccountButtonText,
      'login.button.signInUppercase': signInButtonText,
      'login.button.signIn': pageTitle,
      'customStyles.masqueradeButton': masqueradeButton,
      'customStyles.addQuoteBtn': addQuoteBtn,
      'customStyles.floatingAction': floatingAction,
      'customStyles.addToAllQuoteBtn': addToAllQuoteBtn,
      'customStyles.shoppingListBtn': shoppingListBtn,
    })
      .filter(([, value]) => value)
      .reduce(
        (acc, [translationKey, customizedText]) => ({
          ...acc,
          [translationKey]: customizedText,
        }),
        {},
      ),
  );

  return <LangProvider customText={customText}>{children}</LangProvider>;
}

export default {
  LangWrapper,
};
