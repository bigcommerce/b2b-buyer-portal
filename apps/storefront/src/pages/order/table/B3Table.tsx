import { ChangeEvent, MouseEvent, ReactElement, ReactNode, useContext } from 'react';
import { useB3Lang } from '@b3/lang';
import {
  Card,
  Grid,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TablePagination,
  TableRow,
} from '@mui/material';
import TableSortLabel from '@mui/material/TableSortLabel';

import { b3HexToRgb, getContrastColor } from '@/components/outSideComponents/utils/b3CustomStyles';
import { useMobile } from '@/hooks';
import { CustomStyleContext } from '@/shared/customStyleButton';

import B3NoData from './B3NoData';

interface NodeWrapper<T extends object> {
  node: T;
}

export type PossibleNodeWrapper<T extends object> = T | NodeWrapper<T>;

export const isNodeWrapper = <T extends object>(
  item: PossibleNodeWrapper<T>,
): item is NodeWrapper<T> => 'node' in item;

export type WithRowControls<T> = T & {
  id?: string | number;
};

export interface Pagination {
  offset: number;
  first: number;
  count: number;
}

interface OrderIdRow {
  orderId: string;
}

export interface TableColumnItem<Row extends OrderIdRow> {
  key: string;
  title: string;
  align?: 'right';
  width?: string;
  render: (row: Row) => ReactNode;
  isSortable?: boolean;
}

interface RowProps<Row extends OrderIdRow> {
  columnItems: TableColumnItem<Row>[];
  node: WithRowControls<Row>;
  onClickRow: () => void;
}

function Row<Row extends OrderIdRow>({ columnItems, node, onClickRow }: RowProps<Row>) {
  return (
    <TableRow
      hover
      onClick={onClickRow}
      sx={{
        cursor: 'pointer',
      }}
      data-testid="tableBody-Row"
    >
      {columnItems.map((column) => (
        <TableCell
          align={column.align ?? 'left'}
          key={column.title}
          data-testid={column.key ? `tableBody-${column.key}` : ''}
        >
          {column.render(node)}
        </TableCell>
      ))}
    </TableRow>
  );
}

interface TableProps<Row extends OrderIdRow> {
  columnItems: TableColumnItem<Row>[];
  listItems: WithRowControls<Row>[];
  onPaginationChange?: (pagination: Pagination) => void;
  pagination?: Pagination;
  renderItem: (row: Row, index: number) => ReactElement;
  isInfiniteScroll?: boolean;
  onClickRow: (row: Row, index: number) => void;
  sortDirection?: 'asc' | 'desc';
  sortByFn?: (key: string) => void;
  orderBy?: string;
}

export function B3Table<Row extends OrderIdRow>({
  columnItems,
  listItems = [],
  pagination = {
    offset: 0,
    count: 0,
    first: 10,
  },
  onPaginationChange = () => {},
  renderItem,
  isInfiniteScroll = false,
  onClickRow,
  sortDirection = 'asc',
  sortByFn,
  orderBy = '',
}: TableProps<Row>) {
  const rowsPerPageOptions = [10, 20, 30];

  const {
    state: {
      portalStyle: { backgroundColor = '#FEF9F5' },
    },
  } = useContext(CustomStyleContext);

  const customColor = getContrastColor(backgroundColor);

  const [isMobile] = useMobile();

  const b3Lang = useB3Lang();

  const { offset, count, first } = pagination;

  const handlePaginationChange = (pagination: Pagination) => {
    onPaginationChange(pagination);
  };

  const handleChangePage = (_: MouseEvent<HTMLButtonElement> | null, page: number) => {
    handlePaginationChange({
      ...pagination,
      offset: page * first,
    });
  };

  const handleChangeRowsPerPage = (event: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
    handlePaginationChange({
      ...pagination,
      offset: 0,
      first: parseInt(event.target.value, 10) || first,
    });
  };

  return listItems.length > 0 ? (
    <>
      {isInfiniteScroll && (
        <>
          <Grid container spacing={2}>
            {listItems.map((row, index) => {
              return (
                <Grid item xs={12} key={row.orderId}>
                  {renderItem(row, index)}
                </Grid>
              );
            })}
          </Grid>
          <TablePagination
            rowsPerPageOptions={rowsPerPageOptions}
            labelRowsPerPage={b3Lang('global.pagination.perPage')}
            component="div"
            sx={{
              color: isMobile ? b3HexToRgb(customColor, 0.87) : 'rgba(0, 0, 0, 0.87)',
              marginTop: '1.5rem',
              '::-webkit-scrollbar': {
                display: 'none',
              },
              '& svg': {
                color: isMobile ? b3HexToRgb(customColor, 0.87) : 'rgba(0, 0, 0, 0.87)',
              },
            }}
            count={count}
            rowsPerPage={first}
            page={first === 0 ? 0 : offset / first}
            onPageChange={handleChangePage}
            onRowsPerPageChange={handleChangeRowsPerPage}
          />
        </>
      )}
      {!isInfiniteScroll && (
        <Card
          sx={{
            height: '100%',
            boxShadow:
              '0px 2px 1px -1px rgb(0 0 0 / 20%), 0px 1px 1px 0px rgb(0 0 0 / 14%), 0px 1px 3px 0px rgb(0 0 0 / 12%)',
          }}
        >
          <TableContainer>
            <Table
              sx={{
                tableLayout: 'initial',
              }}
            >
              <TableHead>
                <TableRow data-testid="tableHead-Row">
                  {columnItems.map((column) => (
                    <TableCell
                      key={column.title}
                      width={column.width}
                      align={column.align ?? 'left'}
                      sortDirection={column.key === orderBy ? sortDirection : false}
                      data-testid={column.key ? `tableHead-${column.key}` : ''}
                    >
                      {column.isSortable ? (
                        <TableSortLabel
                          active={column.key === orderBy}
                          direction={column.key === orderBy ? sortDirection : 'desc'}
                          hideSortIcon={column.key !== orderBy}
                          onClick={() => sortByFn?.(column.key)}
                        >
                          {column.title}
                        </TableSortLabel>
                      ) : (
                        column.title
                      )}
                    </TableCell>
                  ))}
                </TableRow>
              </TableHead>

              <TableBody>
                {listItems.map((row, index) => {
                  const node = isNodeWrapper(row) ? row.node : row;

                  return (
                    <Row
                      key={`row-${node.orderId}`}
                      columnItems={columnItems}
                      node={node}
                      onClickRow={() => onClickRow(node, index)}
                    />
                  );
                })}
              </TableBody>
            </Table>
          </TableContainer>
          <TablePagination
            rowsPerPageOptions={rowsPerPageOptions}
            labelRowsPerPage={b3Lang('global.pagination.rowsPerPage')}
            component="div"
            sx={{
              marginTop: '1.5rem',
              '::-webkit-scrollbar': {
                display: 'none',
              },
            }}
            count={count}
            rowsPerPage={first}
            page={first === 0 ? 0 : offset / first}
            onPageChange={handleChangePage}
            onRowsPerPageChange={handleChangeRowsPerPage}
          />
        </Card>
      )}
    </>
  ) : (
    <B3NoData />
  );
}
