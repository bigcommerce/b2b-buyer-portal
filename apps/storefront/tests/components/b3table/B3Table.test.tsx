import { LangProvider } from '@b3/lang';

import { B3Table } from '@/components';
import { TableColumnItem } from '@/components/table/B3Table';
import lang from '@/store/slices/lang';

import { renderWithProviders } from '../../test-utils';

const columnItems: TableColumnItem<1> = {
  key: 'key test1',
  title: 'title test',
};

const listItems = [
  {
    node: {
      id: 1,
    },
  },
];

describe('B3Table component', async () => {
  it('renders correctly', () => {
    renderWithProviders(
      <LangProvider customText={{ 'global.pagination.of': 'test-of' }}>
        <B3Table
          tableFixed
          columnItems={[columnItems]}
          listItems={listItems}
          pagination={{
            offset: 0,
            count: 0,
            first: 10,
          }}
          rowsPerPageOptions={[10, 20, 50]}
          showPagination
          labelRowsPerPage="cards per page"
        />
      </LangProvider>,
      {
        reducer: { lang },
      },
    );
  });

  it('matching b3table with snapshot', () => {
    const { result } = renderWithProviders(
      <LangProvider customText={{ 'global.pagination.of': 'test-of' }}>
        <B3Table
          tableFixed
          columnItems={[columnItems]}
          listItems={listItems}
          pagination={{
            offset: 0,
            count: 0,
            first: 10,
          }}
          rowsPerPageOptions={[10, 20, 50]}
          showPagination
          labelRowsPerPage="cards per page"
          isInfiniteScroll={false}
          isCustomRender={false}
        />
      </LangProvider>,
      {
        reducer: { lang },
      },
    );
    expect(result).toMatchSnapshot();
  });
});
