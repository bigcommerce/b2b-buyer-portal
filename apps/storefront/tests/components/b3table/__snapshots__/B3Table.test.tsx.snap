// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`B3Table component > matching b3table with snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  height: 100%;
  box-shadow: 0px 2px 1px -1px rgb(0 0 0 / 20%),0px 1px 1px 0px rgb(0 0 0 / 14%),0px 1px 3px 0px rgb(0 0 0 / 12%);
}

.emotion-1 {
  width: 100%;
  overflow-x: auto;
}

.emotion-2 {
  display: table;
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
  table-layout: fixed;
}

.emotion-2 caption {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  padding: 16px;
  color: rgba(0, 0, 0, 0.6);
  text-align: left;
  caption-side: bottom;
}

.emotion-3 {
  display: table-header-group;
}

.emotion-4 {
  color: inherit;
  display: table-row;
  vertical-align: middle;
  outline: 0;
}

.emotion-4.MuiTableRow-hover:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

.emotion-4.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-4.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

.emotion-5 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.5rem;
  letter-spacing: 0.01071em;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  color: rgba(0, 0, 0, 0.87);
}

.emotion-6 {
  display: table-row-group;
}

.emotion-8 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  color: rgba(0, 0, 0, 0.87);
  border-bottom: 1px solid rgba(224, 224, 224, 1);
}

.emotion-9 {
  overflow: auto;
  color: rgba(0, 0, 0, 0.87);
  font-size: 0.875rem;
  margin-top: 1.5rem;
}

.emotion-9:last-child {
  padding: 0;
}

.emotion-9::-webkit-scrollbar {
  display: none;
}

.emotion-10 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 56px;
  min-height: 52px;
  padding-right: 2px;
}

@media (min-width:600px) {
  .emotion-10 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:0px) {
  @media (orientation: landscape) {
    .emotion-10 {
      min-height: 48px;
    }
  }
}

@media (min-width:600px) {
  .emotion-10 {
    min-height: 64px;
  }
}

@media (min-width:0px) and (orientation: landscape) {
  .emotion-10 {
    min-height: 52px;
  }
}

@media (min-width:600px) {
  .emotion-10 {
    min-height: 52px;
    padding-right: 2px;
  }
}

.emotion-10 .MuiTablePagination-actions {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-left: 20px;
}

.emotion-11 {
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
}

.emotion-12 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-13 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: inherit;
  font-size: inherit;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 32px;
  margin-left: 8px;
}

.emotion-13.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-13 .MuiTablePagination-select {
  padding-left: 8px;
  padding-right: 24px;
  text-align: right;
  -webkit-text-align-last: right;
  text-align-last: right;
}

.emotion-14 {
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 0;
  cursor: pointer;
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-14:focus {
  background-color: rgba(0, 0, 0, 0.05);
  border-radius: 0;
}

.emotion-14::-ms-expand {
  display: none;
}

.emotion-14.Mui-disabled {
  cursor: default;
}

.emotion-14[multiple] {
  height: auto;
}

.emotion-14:not([multiple]) option,
.emotion-14:not([multiple]) optgroup {
  background-color: #fff;
}

.emotion-14.emotion-14.emotion-14 {
  padding-right: 24px;
  min-width: 16px;
}

.emotion-14.MuiSelect-select {
  height: auto;
  min-height: 1.4375em;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-14::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-14::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-14:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-14::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-14:focus {
  outline: 0;
}

.emotion-14:invalid {
  box-shadow: none;
}

.emotion-14::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-14::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-14::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-14:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-14::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-14:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-14:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-14:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-14:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-14.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-14:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-15 {
  bottom: 0;
  left: 0;
  position: absolute;
  opacity: 0;
  pointer-events: none;
  width: 100%;
  box-sizing: border-box;
}

.emotion-16 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  position: absolute;
  right: 0;
  top: calc(50% - .5em);
  pointer-events: none;
  color: rgba(0, 0, 0, 0.54);
}

.emotion-16.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-17 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-18 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: inherit;
}

.emotion-18::-moz-focus-inner {
  border-style: none;
}

.emotion-18.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-18 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-18:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-18:hover {
    background-color: transparent;
  }
}

.emotion-18.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-19 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

<div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div
      class="MuiTableContainer-root emotion-1"
    >
      <table
        class="MuiTable-root emotion-2"
      >
        <thead
          class="MuiTableHead-root emotion-3"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head emotion-4"
            data-testid="tableHead-Row"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium emotion-5"
              data-testid="tableHead-key test1"
              scope="col"
            >
              title test
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root emotion-6"
        >
          <tr
            class="MuiTableRow-root emotion-4"
            data-testid="tableBody-Row"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-8"
              data-testid="tableBody-key test1"
            />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root emotion-9"
      of="test-of"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar emotion-10"
      >
        <div
          class="MuiTablePagination-spacer emotion-11"
        />
        <p
          class="MuiTablePagination-selectLabel emotion-12"
          id=":rX:"
        >
          cards per page
        </p>
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input emotion-13"
        >
          <div
            aria-controls=":rX:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby=":rX: :rX:"
            class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input emotion-14"
            id=":rX:"
            role="combobox"
            tabindex="0"
          >
            10
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput emotion-15"
            tabindex="-1"
            value="10"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard emotion-16"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTablePagination-displayedRows emotion-17"
        >
          0–0 of 0
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium emotion-18"
            disabled=""
            tabindex="-1"
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-19"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium emotion-18"
            disabled=""
            tabindex="-1"
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-19"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
